docs/versioning.md

```diff
@@ -0,0 +1,29 @@
+### Assumptions
+This project uses [semver v2](https://semver.org/), as does the rest of OpenTelemetry.
+
+### Goals
+#### API Stability:
+* Once the API for a given signal (spans, logs, metrics, baggage) has been officially released, that API module will function, with any SDK that has the same major version, and equal or greater minor or patch version.
+  * example: libraries that are instrumented with opentelemetry-api-trace:1.0.1 will function with SDK library opentelemetry-sdk-trace:1.11.33.
+* No existing method names or signatures will change with patch versioning.  We will only change signatures in a backwards compatible way with minor versioning.
+
+#### SDK Stability:
+* Public portions of the SDK (constructors, configuration, end-user interfaces) must remain backwards compatible.
+* Internal interfaces are allowed to break.
+* No existing method names or signatures will change with patch versioning.  We will only change signatures in a backwards compatible way with minor versioning.
+
+### Methods
+#### Mature signals
+* Public portions of the SDK (constructors, configuration, end-user interfaces) must remain backwards compatible.
+* Internal interfaces are allowed to break.
```

Comment:
Break when? What defines an internal interface?

Reply:
In PHP:

from [interfaces](https://www.php.net/manual/en/language.oop5.interfaces.php):
Interfaces are defined in the same way as a class, but with the interface keyword replacing the class keyword and without any of the methods having their contents defined.

Internal interfaces would be defined to me as a non-client facing part of the API.  Perhaps this bullet point isn't necessary.

Reply:
I still think the difference between internal and otherwise (public? external?) isn't specifically defined in the document. Is it impossible for a consuming client to reach them? If so, I think it's irrelevant for this doc. If not, then it should be made clear to consumers which interfaces are meant to be internal (and therefore may break with non-major versions). I may be missing some php-specific piece of lore here though.